src/analysis/imports.rs

```diff
@@ -231,27 +253,27 @@ impl Imports {
                 self.add_with_version(&used_type[..i], version);
             }
         } else {
-            self.add_with_version(used_type, version);
+            self.add_with_version(&format!("crate::{}", used_type), version);
         }
     }
 
     /// Tries to strip crate name prefix from given name.
     ///
     /// Returns `None` if name matches crate name exactly. Otherwise returns
     /// name with crate name prefix stripped or full name if there was no match.
-    fn strip_crate_name<'a>(&self, name: &'a str) -> Option<&'a str> {
+    fn strip_crate_name<'a>(&self, name: &'a str) -> Option<Cow<'a, str>> {
         let prefix = &self.crate_name;
         if !name.starts_with(prefix) {
-            return Some(name);
+            return Some(Cow::Borrowed(name));
         }
         let rest = &name[prefix.len()..];
         if rest.is_empty() {
             None
-        } else if let Some(prefix) = rest.strip_prefix("::") {
-            Some(prefix)
+        } else if rest.starts_with("::") {
```

Comment:
Where would `::Foo` come from? Let's maybe just not generate that anymore and directly have it create `crate::Foo` instead?

Reply:
Well, from the current crate obviously. :smile:

Reply:
Why would someone pass `::Foo` in here then instead of directly going with `crate::Foo`?

Reply:
No one. The point of this function is to strip the crate name. So if you receive "glib::Foo", `rest` will be "::Foo".

Reply:
Ok then